server/devices/Device.cc

```diff
@@ -1,47 +1,49 @@
-#include "Device.h"
-
-namespace smart_home {
-    Device::Device(DeviceType deviceType) : type(deviceType) {}
-
-    std::string Device::getDeviceType() const {
-        switch (type) {
-            case LIGHT:
-                return "Light";
-            case HEATER:
-                return "Heater";
-            default:
-                return "ERROR";
-        }
-    }
-
-    std::string Device::getDeviceStatus() const {
-        if (status == 1)
-            return "On";
-        else if (status == 0)
-            return "Off";
-        else
-            return "ERROR";
-    }
-
-    void Device::setDeviceStatus(int status) {
-        if (status > 1 || status < 0){
-            std::cout << "Cannot set this status, command ignored!" << std::endl;
-            return; 
-        }
-        this->status = status;
-    }
-
-    void Device::setDeviceType(DeviceType deviceType) {
-        this->type = deviceType;
-    }
-
-    void Device::turnOn() {
-        std::cout << "Turning on " << getDeviceType() << std::endl;
-        status = 1;
-    }
-
-    void Device::turnOff() {
-        std::cout << "Turning off " << getDeviceType() << std::endl;
-        status = 0;
-    }
+#include "Device.h"
+
+namespace smart_home {
+    Device::Device(DeviceType deviceType) : type(deviceType) {}
+
+    std::string Device::getDeviceType() const {
+        switch (type) {
+            case LIGHT:
+                return "Light";
+            case HEATER:
+                return "Heater";
+            case AIR_CONDITIONER:
+                return "Air Conditioner";
+            default:
+                return "ERROR";
+        }
+    }
+
+    std::string Device::getDeviceStatus() const {
+        if (status == 1)
+            return "On";
+        else if (status == 0)
+            return "Off";
+        else
+            return "ERROR";
+    }
+
+    void Device::setDeviceStatus(int status) {
+        if (status > 1 || status < 0){
```

Comment:
a cleaner way to handle this status types would be to use enum class. if you're not familiar with it, have a quick read please. Imagine you'd have multiple possible statuses(STARTING, SERVICE, UPDATE, etc), somebody would have to search through the code to see what each number means.

Reply:
later edit, I've seen you have used enum class in other parts of your homework